packages/create/src/cli/change-data-source/change-data-source.ts

```diff
@@ -0,0 +1,152 @@
+import type { Command } from 'commander';
+import type { AdpProjectData, DataSourceData, PromptDefaults, ChangeDataSourceAnswers } from '@sap-ux/adp-tooling';
+import type { YUIQuestion } from '@sap-ux/inquirer-common';
+import {
+    generateChange,
+    ChangeType,
+    getPromptsForChangeDataSource,
+    getTargetDataSources,
+    getDataSourcesDictionary
+} from '@sap-ux/adp-tooling';
+import { createAbapServiceProvider } from '@sap-ux/system-access';
+import { getLogger, traceChanges } from '../../tracing';
+import prompts from 'prompts';
+import { readFileSync, existsSync } from 'fs';
+import { join } from 'path';
+import { UI5Config } from '@sap-ux/ui5-config';
+import { promptYUIQuestions } from '../../common';
+
+let loginAttempts = 3;
+
+/**
+ * Add a new sub-command to change the data source of an adaptation project to the given command.
+ *
+ * @param {Command} cmd - The command to add the change-data-source sub-command to.
+ */
+export function addChangeDataSourceCommand(cmd: Command): void {
+    cmd.command('change-data-source [path]')
+        .option('-s, --simulate', 'simulate only do not write or install')
+        .action(async (path, options) => {
+            await changeDataSource(path, { ...options }, !!options.simulate);
+        });
+}
+
+/**
+ * Changes the data source of an adaptation project.
+ *
+ * @param {string} basePath - The path to the adaptation project.
+ * @param {PromptDefaults} defaults - The default values for the prompts.
+ * @param {boolean} simulate - if set to true, then no files will be written to the filesystem
+ */
+async function changeDataSource(basePath: string, defaults: PromptDefaults, simulate: boolean): Promise<void> {
+    const logger = getLogger();
+    try {
+        if (!basePath) {
+            basePath = process.cwd();
+        }
+        const configJsonPath = join(basePath, '.adp', 'config.json');
+        if (existsSync(configJsonPath)) {
+            const config = JSON.parse(readFileSync(configJsonPath, 'utf-8'));
+            if (config.environment === 'CF') {
+                throw new Error('Changing data source is not supported for CF projects.');
+            }
+        }
+
+        const variant = JSON.parse(readFileSync(join(basePath, 'webapp', 'manifest.appdescr_variant'), 'utf-8'));
+        const ui5Config = await UI5Config.newInstance(readFileSync(join(basePath, 'ui5.yaml'), 'utf-8'));
+        const { destination, url, client } =
+            ui5Config.findCustomMiddleware<{ backend: Array<{ destination?: string; url?: string; client?: string }> }>(
+                'fiori-tools-proxy'
```

Comment:
You should use the preview middleware instead because thats the middleware that is used during a preview to merge the appdescr with the manifest

Reply:
Fixed with: b335f48373d6f448c5f9ef6201e220ed30bbca88